monai/transforms/post/array.py

```diff
@@ -182,52 +184,74 @@ class KeepLargestConnectedComponent(Transform):
     """
 
     def __init__(
-        self, applied_values, independent: bool = True, background: int = 0, connectivity: Optional[int] = None
+        self, applied_labels=None, independent: bool = True, background: int = 0, connectivity: Optional[int] = None
     ):
         """
         Args:
-            applied_values (list or tuple of int): number list for applying the connected component on.
-                The pixel whose value is not in this list will remain unchanged.
-            independent: consider several labels as a whole or independent, default is `True`.
+            applied_labels (int, list or tuple of int): Labels for applying the connected component on.
+                If only one channel. The pixel whose value is not in this list will remain unchanged.
+                If the data is in one-hot format, this is used to determine what channels to apply.
+            independent (bool): consider several labels as a whole or independent, default is `True`.
                 Example use case would be segment label 1 is liver and label 2 is liver tumor, in that case
                 you want this "independent" to be specified as False.
             background: Background pixel value. The over-segmented pixels will be set as this value.
+                Note that for one-hot data, the over-segment pixels will be set to 0 in its channel.
             connectivity: Maximum number of orthogonal hops to consider a pixel/voxel as a neighbor.
                 Accepted values are ranging from  1 to input.ndim. If ``None``, a full
                 connectivity of ``input.ndim`` is used.
         """
         super().__init__()
-        self.applied_values = applied_values
+        if applied_labels is None:
+            raise ValueError("Please provide applied_labels.")
+
+        self.applied_labels = ensure_tuple(applied_labels)
         self.independent = independent
         self.background = background
         self.connectivity = connectivity
-        if background in applied_values:
-            raise ValueError("Background pixel can't be in applied_values.")
+
+        if background in self.applied_labels:
```

Comment:
If the input data is in One-Hot format and you want to apply this transform for `channel 0`, it will raise an error. I think this sanity check is unnecessary here.

Reply:
If the input data is not in one-hot,
previously we say want to support user to set the background.
The background here means that the over-segment pixels will be set to this value.
But this is only true when the data is not in one-hot.

I think this check is necessary to support that scenario.
As if user say applied_labels=[1,2] and say background to 1
Without this check, With input data of 1 channel,  the over-segment part will all be set to 1.
Which I think then make this transform useless/not correct.
[[0,0,2],
 [0,1,2], 
 [2,1,2]]
   if we have independent = True, applied_labels=[1,2], background = 1
we will get
[[0,0,2],
 [0,1,2], 
 [1,1,2]]
which I think is not correct, so the check is necessary.

I propose that we remove this variable.
And in the case of 1 channel input data, always treat 0 as background and set over-segment pixels to 0.
As it kind of will be more consistent to if the input data is of one-hot, the over-segment pixel we are setting them to 0 at that channel.

Reply:
OK, let's remove this arg for current version. If someday users raise the request for it, we can add it back and will not impact existing logic.
Thanks.